src/in_memory_time_series_storage.jl

```diff
@@ -116,12 +116,12 @@ function deserialize_time_series(
 end
 
 function deserialize_time_series(
-    ::Type{<:AbstractDeterministic},
+    ::Type{T},
     storage::InMemoryTimeSeriesStorage,
     ts_metadata::TimeSeriesMetadata,
     rows::UnitRange,
     columns::UnitRange,
-)
+) where T <: TimeSeriesData
```

Comment:
This will fail if the user passes `AbstractDeterministic`.  You could check for that and re-map it to Deterministic.